data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template

```diff
@@ -152,11 +150,31 @@ $IPTABLES -t raw -A DHCP_IRONIC -j DROP
 {{end}}
 
 # Wait for images to be downloaded/ready
-podman wait -i 1000ms ipa-downloader
+#podman wait -i 1000ms ipa-downloader
 podman wait -i 1000ms coreos-downloader
-while ! curl --fail http://localhost/images/rhcos-ootpa-latest.qcow2.md5sum ; do sleep 1; done
-while ! curl --fail --head http://localhost/images/ironic-python-agent.initramfs ; do sleep 1; done
-while ! curl --fail --head http://localhost/images/ironic-python-agent.kernel ; do sleep 1; done
+
+# Write master network config to /tmp/nmstate for image_customization_controller to serve our merged images
+mkdir -p /tmp/nmstate
+{{range .PlatformData.BareMetal.Hosts}}
+    cat >/tmp/nmstate/{{.Name}}.yaml <<EOF
+{{.NetworkConfig}}
+EOF
+{{end}}
+
+CUSTOMIZATION_IMAGE="virthost.ostest.test.metalkube.org:5000/localimages/image-customization-controller"
+
+# Embed agent ignition into the rhcos live iso
+sudo podman run -d --net host --privileged --name image-customization \
+    --env DEPLOY_ISO="/shared/html/images/ironic-python-agent.iso" \
+    --env IRONIC_BASE_URL="http://{{.PlatformData.BareMetal.APIVIP}}" \
+    --env IRONIC_AGENT_IMAGE="" \
+    --env IRONIC_AGENT_PULL_SECRET="" \
```

Comment:
We need to set these to stop it from pulling from @dtantsur's quay.io account, which is temporarily the default.

I had a go at this [here](https://github.com/zaneb/openshift-installer/commit/24cf61e62c5371cd9b5817f2d98b36be5a3f33a1) but it turns out that, at least in dev-scripts, the pull secret we have is for the mirror (virthost.ostest.test.metalkube.org:5000). So I think we need to add the file https://github.com/openshift/installer/blob/master/data/data/bootstrap/files/etc/containers/registries.conf.template to the ignition in image-customization-controller.
I'm not clear on how this will work once we are running in the cluster. Presumably for disconnected installs at least there will be something similar. Where do we get the mirror list from in that case? @hardys, do you know?

Reply:
https://github.com/openshift/image-customization-controller/pull/20 will enable us to embed the registries.conf file.

Reply:
@zaneb when the image exists in the release payload we should just be able to use `image_for` like the other images at the top of this script - we shouldn't need to care about disconnected or not.

We pass a reference to the virthost registry via [imageContentSources in the install-config](https://github.com/openshift-metal3/dev-scripts/blob/master/utils.sh#L382), and that ends up templated into registries.conf already.

Reply:
More info after re-reading the question :)  The same registries.conf content ends up on the running cluster:

```
[root@master-0 ~]# cat /etc/containers/registries.conf
unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]
short-name-mode = ""

[[registry]]
  prefix = ""
  location = "quay.io/openshift-release-dev/ocp-v4.0-art-dev"
  mirror-by-digest-only = true

  [[registry.mirror]]
    location = "virthost.ostest.test.metalkube.org:5000/localimages/local-release-image"

[[registry]]
  prefix = ""
  location = "registry.ci.openshift.org/ocp/release"
  mirror-by-digest-only = true

  [[registry.mirror]]
    location = "virthost.ostest.test.metalkube.org:5000/localimages/local-release-image"
```

That gets created via ignition, from the MCS rendered content:

```
    "filesystem": "root",
    "overwrite": true,
    "path": "/etc/containers/registries.conf",
    "contents": {
...
```
Looks like that comes from the `01-master-container-runtime` MachineConfig object (not entirely sure how that's populated though, I'm guessing it copies the bootstrap file since I don't see it populated via installer manifests).

Reply:
Ah, if you override images then they point to virthost.ostest.test.metalkube.org (which works), so that is possibly how we tested.
If you don't then they point to teh Internets, so we'll need to pass the registries.conf file in the IPA Ignition in the cluster as well as on the bootstrap.

Reply:
That patch looks like it got messed up. https://github.com/zaneb/openshift-installer/commit/73d656816d1491f12e07f3387357fda464df5d80 is a rebased one.